src/core/geometry/qgscircularstring.h

```diff
@@ -82,6 +82,7 @@ class CORE_EXPORT QgsCircularString: public QgsCurve
     QDomElement asGml3( QDomDocument &doc, int precision = 17, const QString &ns = "gml", QgsAbstractGeometry::AxisOrder axisOrder = QgsAbstractGeometry::AxisOrder::XY ) const override;
     json asJsonObject( int precision = 17 ) const override SIP_SKIP;
     bool isEmpty() const override SIP_HOLDGIL;
+    bool isValid( QString &error SIP_OUT, int flags = 0 ) const override SIP_HOLDGIL;
```

Comment:
we shouldn't be holding the gil for these calls - it can call out to geos which may take some time, and we should only ever hold the gil for trivial/very inexpensive calls